### PR TITLE
add posibility to use existed security group that can be used in work…

### PR DESCRIPTION
**Pull request by dmitriy-lukyanchikov**
1) add possibility to use existing security group
2) add possibility to add additional list of security groups to autoscaling group

@aknysh please review )

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -148,7 +149,7 @@ module "autoscale_group" {
 
   image_id                  = "${var.use_custom_image_id == "true" ? var.image_id : join("", data.aws_ami.eks_worker.*.id)}"
   iam_instance_profile_name = "${local.use_existing_instance_profile == "false" ? join("", aws_iam_instance_profile.default.*.name) : var.aws_iam_instance_profile_name}"
-  security_group_ids        = ["${join("", aws_security_group.default.*.id)}"]
+  security_group_ids        = ["${local.use_existing_workers_security_group == "false" ? join("", aws_security_group.default.*.id) : var.workers_security_group_id}", "${var.additional_security_group_ids}"]
```
Comment (aknysh) [inline main.tf]:
```suggestion
  security_group_ids        = ["${compact(concat(local.use_existing_workers_security_group == "false" ? join("", aws_security_group.default.*.id) : var.workers_security_group_id}", "${var.additional_security_group_ids}))"]
```

**Comment by dmitriy-lukyanchikov** [comment]
add suggested changes to the branch @aknysh

File: main.tf [hunk 2]
```diff
@@ -148,7 +149,7 @@ module "autoscale_group" {
 
   image_id                  = "${var.use_custom_image_id == "true" ? var.image_id : join("", data.aws_ami.eks_worker.*.id)}"
   iam_instance_profile_name = "${local.use_existing_instance_profile == "false" ? join("", aws_iam_instance_profile.default.*.name) : var.aws_iam_instance_profile_name}"
-  security_group_ids        = ["${join("", aws_security_group.default.*.id)}"]
+  security_group_ids        = ["${compact(concat(local.use_existing_workers_security_group == "false" ? join("", aws_security_group.default.*.id) : var.workers_security_group_id}", "${var.additional_security_group_ids}))"]
```
Comment (aknysh) [inline main.tf]:
```suggestion
    security_group_ids        = ["${compact(concat(list(local.use_existing_workers_security_group == "false" ? join("", aws_security_group.default.*.id) : var.workers_security_group_id), var.additional_security_group_ids))}"]

```

Comment (aknysh) [reply main.tf]: please check the CI/CD for errors
https://travis-ci.org/cloudposse/terraform-aws-eks-workers/builds/529627667?utm_source=github_status&utm_medium=notification

**Comment by dmitriy-lukyanchikov** [comment]
@aknysh thank you miss the error, also delete variable use_existing_image_id and use local to get false or true for further count calculation

**Comment by dmitriy-lukyanchikov** [comment]
@aknysh @osterman i think its ready to final review

**Comment by dmitriy-lukyanchikov** [comment]
do not merge it looks like using locals to detect existing resources are not working and i got value of 'count' cannot be computed on last check so we need some other solution here @aknysh 
i propose to use additional variable as it as for image_id before, what d you think?

**Comment by aknysh** [comment]
@dmitriy-lukyanchikov 
yea, the infamous `count` error.
it's better to add the `use_existing_image_id` variable back than to introduce the possibility of the count errors.

**Comment by dmitriy-lukyanchikov** [comment]
add variables instead of locals for using exisitng instance profile and security groups, also add possibility to attach additional policy to worker roles usign variab le `workers_role_policy_arns`

**Comment by dmitriy-lukyanchikov** [comment]
@aknysh can you please review the code

**Comment by dmitriy-lukyanchikov** [comment]
@aknysh @osterman can you review and merge this changes, its really needed

File: variables.tf [hunk 3]
```diff
@@ -396,3 +396,33 @@ variable "aws_iam_instance_profile_name" {
...
+variable "use_existing_aws_iam_instance_profile" {
+  type        = "string"
+  description = "If set to `true`, will use variable `aws_iam_instance_profile_name` to run EKS workers using existing aws instance profile that was created outside this module, workaround to not get error like `count cannot be computed`"
+  default     = "false"
+}
+
+variable "workers_role_policy_arns" {
+  type        = "list"
+  default     = []
+  description = "List of role polcicies taht will be attached to workers default role on creation"
```
Comment (aknysh) [inline variables.tf]:
```suggestion
  description = "List of role policies that will be attached to workers default role on creation"
```

**Comment by dmitriy-lukyanchikov** [comment]
@aknysh thank you, fix typo

File: variables.tf [hunk 4]
```diff
@@ -396,3 +396,33 @@ variable "aws_iam_instance_profile_name" {
...
+variable "use_existing_aws_iam_instance_profile" {
+  type        = "string"
+  description = "If set to `true`, will use variable `aws_iam_instance_profile_name` to run EKS workers using existing aws instance profile that was created outside this module, workaround to not get error like `count cannot be computed`"
+  default     = "false"
+}
+
+variable "workers_role_policy_arns" {
+  type        = "list"
+  default     = []
+  description = "List of role polcicies that will be attached to workers default role on creation"
```
Comment (aknysh) [inline variables.tf]:
```suggestion
  description = "List of policy ARNs that will be attached to the workers default role on creation"
```
sorry, made a typo as well :)

File: variables.tf [hunk 5]
```diff
@@ -396,3 +396,33 @@ variable "aws_iam_instance_profile_name" {
   default     = ""
   description = "The name of the existing instance profile that will be used in autoscaling group for EKS workers. If empty will create a new instance profile."
 }
+
+variable "workers_security_group_id" {
+  type        = "string"
+  default     = ""
+  description = "The name of the existing security group that will be used in autoscaling group for EKS workers. If empty will create a new security group."
```
Comment (aknysh) [inline variables.tf]:
```suggestion
  description = "The name of the existing security group that will be used in autoscaling group for EKS workers. If empty, a new security group will be created"
```

File: variables.tf [hunk 6]
```diff
@@ -396,3 +396,33 @@ variable "aws_iam_instance_profile_name" {
...
+}
+
+variable "use_existing_security_group" {
+  type        = "string"
+  description = "If set to `true`, will use variable `workers_security_group_id` to run EKS workers using existing security group that was created outside this module, workaround to not get error like `count cannot be computed`"
```
Comment (aknysh) [inline variables.tf]:
```suggestion
  description = "If set to `true`, will use variable `workers_security_group_id` to run EKS workers using the existing security group that was created outside of this module, workaround for errors like `count cannot be computed`"
```

File: variables.tf [hunk 7]
```diff
@@ -396,3 +396,33 @@ variable "aws_iam_instance_profile_name" {
...
+  default     = "false"
+}
+
+variable "additional_security_group_ids" {
+  type        = "list"
+  default     = []
+  description = "Additional list of security groups that will be attached to autoscaling group."
```
Comment (aknysh) [inline variables.tf]:
```suggestion
  description = "Additional list of security groups that will be attached to the autoscaling group"
```

File: variables.tf [hunk 8]
```diff
@@ -396,3 +396,33 @@ variable "aws_iam_instance_profile_name" {
...
+}
+
+variable "use_existing_aws_iam_instance_profile" {
+  type        = "string"
+  description = "If set to `true`, will use variable `aws_iam_instance_profile_name` to run EKS workers using existing aws instance profile that was created outside this module, workaround to not get error like `count cannot be computed`"
```
Comment (aknysh) [inline variables.tf]:
```suggestion
  description = "If set to `true`, will use variable `aws_iam_instance_profile_name` to run EKS workers using an existing AWS instance profile that was created outside of this module, workaround for error like `count cannot be computed`"
```

**Comment by aknysh** [comment]
@dmitriy-lukyanchikov  a few more changes

**Comment by dmitriy-lukyanchikov** [comment]
@aknysh all fixed

**Pull request merged**